### DISCUSSION TITLE
Issue scaling FastAPI on cloud run

### DISCUSSION THREAD

**Discussion by louis030195** [fastapi/fastapi#5927] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
FROM python:3.10-slim

ENV PYTHONUNBUFFERED True

ENV APP_HOME /app
WORKDIR $APP_HOME
COPY . ./

ENV PORT 8080

RUN pip install --no-cache-dir -r requirements.txt

CMD exec gunicorn -w 1 -k uvicorn.workers.UvicornWorker main:app -b 0.0.0.0:${PORT} --threads 80 --timeout 0 --log-level info
```

### Description

Hi, I'm having lag issue scaling FastAPI on Google Cloud Run

<img width="1395" alt="image" src="https://user-images.githubusercontent.com/25003283/214899298-6aa1e6a3-0ff5-4b58-81d4-2d5ae6a5c8b0.png">

<img width="1396" alt="image" src="https://user-images.githubusercontent.com/25003283/214899365-66b4b75c-05d2-4f45-ab2d-f54c001206c1.png">

### Operating System

Linux

### Operating System Details

![image](https://user-images.githubusercontent.com/25003283/214900325-2670d756-7429-41bf-afc6-a1d6544f42ff.png)

![image](https://user-images.githubusercontent.com/25003283/214900297-46f4f365-fcd5-47ab-9da1-8afb47b11f1b.png)

### FastAPI Version

latest

### Python Version

3.10

### Additional Context

Main endpoints are doing SSE stream using `sse-starlette` and `EventSourceResponse`, I wonder how it is handled concurrently by FastAPI?

Thanks!

**Comment by caeser1996** [comment]
FastAPI uses the Starlette framework for serving asynchronous requests, including Server-Sent Events (SSE). SSE streams are handled concurrently in FastAPI using the asynchronous programming model, allowing multiple clients to receive updates from the server at the same time without blocking the event loop.

If you're encountering lag issues when scaling FastAPI on Google Cloud Run, it could be due to several reasons, including:

Network latency
Insufficient resource allocation on the host machine
Unoptimized application code
I would recommend profiling your application to identify the root cause of the lag and considering optimizing the relevant parts of your code, or adjusting the resources allocated to the host machine.
